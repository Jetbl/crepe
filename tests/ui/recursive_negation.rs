// This test has a negated clause, but it is not stratified.

use crepe::crepe;

crepe! {
    struct Ok();

    Ok() <- !Ok();
}

fn main() {}
