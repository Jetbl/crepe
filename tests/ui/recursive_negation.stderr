error: Negation of relation 'Ok' creates a dependency cycle and cannot be stratified.
 --> $DIR/recursive_negation.rs:8:14
  |
8 |     Ok() <- !Ok();
  |              ^^
